### PR TITLE
GH Actions: Update dependencies, drop Ubuntu 18.04, clean up

### PR DESCRIPTION
**Pull request by Geod24**
Running the action currently triggers 83 warnings, let's see how many we remove with this.

### CODE REVIEW

**Comment by Geod24** [comment]
Scheduled Ubuntu 18.04 brownout, great timing

**Comment by Geod24** [comment]
99% of warnings are gone, this should be G2G

File: .github/workflows/runnable_cxx.yml [hunk 1]
```diff
@@ -56,17 +57,13 @@ jobs:
...
         target: [
-          # Versions of clang earlier than 8 are not available on 18.04
-          # Versions of clang earlier than 11 are not available on 20.04
+          # Versions of clang earlier than 11 are not available on 20.04, but are on macOS-11
           clang-13.0.0, clang-12.0.0, clang-11.0.0, clang-10.0.0, clang-9.0.0, clang-8.0.0,
-          # For g++, we test all major's latest minor releases since the
-          # introduction of ABI the dual ABI (v5.1, 2015-04-15)
-          g++-11, g++-10, g++-9, g++-8, g++-7, g++-6, g++-5,
+        # For g++, we test the oldest compiler on Ubuntu 20.04, which is GCC-9
+          g++-11, g++-10, g++-9,
```
Comment (ibuclaw) [inline .github/workflows/runnable_cxx.yml:66]: No g++-12?

**Review by ibuclaw — APPROVED** [review]

File: .github/workflows/runnable_cxx.yml (hunk 1 above)
Comment (Geod24) [reply .github/workflows/runnable_cxx.yml:66]: I'll look into adding support in another PR

**Pull request merged**